### ISSUE TITLE
Angular2 Typescript version?

### ISSUE THREAD

**Issue by aderbas** [hgoebl/Leaflet.MultiOptionsPolyline#6]
Theoretically it is not an issue, but I'm using Angular2 and I was very interested in using in my project, it's very easy to use. But I can not use with typescript in my code, I try to put MultiOptionsPolyline in Leaflet Typings but not working =/

Some help how to use in Angular2?

**Comment by hgoebl** [comment]
It's not a problem of Angular2, but the ambient TypeScript definitions are not yet present.
I'll put it on my list...

**Comment by aderbas** [comment]
I know, thank you for your attention.

**Comment by aderbas** [comment]
I got it in a not very nice way. I installed typings of Leaflat and edited file index.d.ts :

```
....
 // Try export MultiOptionsPolyline ###################### Aderbal Nunes
export interface MultiOptionsPolyline extends FeatureGroup{
    getLatLngs():LatLng[];
    getLatLngsSegments():LatLng[];
}
export function multiOptionsPolyline(latlng: LatLng[], options?: MarkerOptions): Polyline;
// ###################################################################
....
```
I downloaded the file `Leaflet.MultiOptionsPolyline.js` put in my lib folder and import in angular-cli.json

```
...
"scripts": [
     "../node_modules/leaflet/dist/leaflet.js",
     "../lib/multi-polyline-options.js"
 ]
...
```
It worked perfectly in my project. Of course if I upgrade typings from repo, will stop working. But it's what I do. Thanks!

**Comment by hgoebl** [comment]
Thank you very much for sharing your solution.

Generally it shouldn't be much of problem to externalize the definitions for MultiOptionsPolyline so leaflet.d.ts keeps clean. Unfortunately I don't have any time currently and in the next couple of weeks.

I'm going to create a .d.ts file and make a pull request to DefinitelyTyped. If somebody wants to take this task, please go ahead.

**Comment by hgoebl** [comment]
I haven't used DefinitelyTyped, but added an index.d.ts and mentioned it in package.json/typings.

**Comment by infrarosso78** [comment]
Hi, sorry to reopen this but i get plenty of errors when I try to compile your plugin on Angular 6.
I've added the package with yarn add hgoebl/Leaflet.MultiOptionsPolyline (pretty much the same of npm i --save and during compilation it fails with this messages: 
ERROR in node_modules/Leaflet.MultiOptionsPolyline/index.d.ts(5,60): error TS2304:Cannot find name 'PolylineOptions'.
node_modules/Leaflet.MultiOptionsPolyline/index.d.ts(10,18): error TS2304: Cannot find name 'PolylineOptions'.
node_modules/Leaflet.MultiOptionsPolyline/index.d.ts(16,58): error TS2304: Cannot find name 'MarkerOptions'.
node_modules/Leaflet.MultiOptionsPolyline/index.d.ts(20,51): error TS2304: Cannot find name 'FeatureGroup'.
node_modules/Leaflet.MultiOptionsPolyline/index.d.ts(21,28): error TS2304: Cannot find name 'LatLng'.
...
Basically your index.d.ts cannot retrieve the Leaflet typings
